README.md

```diff
@@ -1,37 +1,64 @@
 
-Event driven microservices promote responsiveness allowing decisions to be made faster. Services also become more resilient to failure as they can quickly rebuild their state by replaying events.
+Streambed is a curated set of dependencies and a toolkit for writing asynchronous event-driven services that aim to run on the 
+smallest "std" targets supported by Rust. Streambed components use a single core MIPS32 OpenWrt device running at around 500MHz 
+and 128MiB as a baseline target.
 
-Efficient
----
+A commit log modelled on Apache Kafka is provided, along with partial and extended support for the Kafka HTTP API.
 
-Streambed based applications are designed to run at the edge on embedded computers as well as in the cloud and so efficient CPU and memory usage are of primary concern.
+A secret stored modelled on Hashicorp Vault is provided along with partial support for the Vault HTTP API.
 
-Secure
----
+Production services using the commit log and secret store have been shown to use less than 3MiB of resident memory
+while also offering good performance.
 
-Security is another primary consideration throughout the design of Streambed. For example, in the world of the Internet of Things, if an individual sensor becomes compromised then its effects can be minimized.
+## Streambed's traits
+### Event-driven services
 
-Built for integration
----
+Event-driven services promotes responsiveness as events can be pushed to where they need to be consumed; by the user
+of a system. Event-driven services are also resilient to failure as they can use "event sourcing" to 
+quickly rebuild their state through replaying events.
+
+### Efficient
+
+Streambed based applications are designed to run at the edge on embedded computers as well as in the cloud and so efficient CPU and memory usage are a primary concern.
+
+### Secure
+
+Security is a primary consideration throughout the design of Streambed. For example, in the world of the Internet of Things, if an individual sensor becomes compromised then its effects can be minimized.
+
+### Built for integration
 
 Streambed is a toolkit that promotes the consented sharing of data between many third-party applications. No more silos of data. Improved data availability leads to better decision making, which leads to better business.
 
-Standing on the shoulders of giants, leveraging existing communities
----
+### Standing on the shoulders of giants, leveraging existing communities
 
 Streambed is an assemblage of proven approaches and technologies that already have strong communities. Should you have a problem there are many people and resources you can call on.
 
-Open source and open standards
----
+### Open source and open standards
 
 Streambed is entirely open source providing cost benefits, fast-time-to-market, the avoidance of vendor lock-in, improved security and more.
 
-Rust
----
+### Rust
 
 Streambed builds on Rust's traits of writing fast and efficient software correctly.
```

Comment:
"traits" may be a confusing word here as that means something specific in the Rust language - perhaps something more Rust-agnostic like "Streambed builds on the strengths of the Rust language, enabling us to write fast and efficient software correctly." Or maybe the use of the word "trait" was on-purpose :)

Reply:
ok, eliminated the use of the word trait.